File: codeWars.py
def get_count(sentence):
    count = 0
    for element in sentence:
        if element=='a' or element=='e' or element=='i' or element=='o' or element=='u':
            count += 1
            
    return count

# Write a function that takes in a string of one or more words,
#  and returns the same string, but with all five or more letter words reversed
#   (Just like the name of this Kata). Strings passed in will consist of only letters and spaces. 
#   Spaces will be included only when more than one word is present.

def spin_words(sentence):
    array = sentence.split()
    reverse_array = []
    for element in array:
        if len(element) >= 5:
            reverse = element[::-1]
            reverse_array.append(reverse)
        else:
            reverse_array.append(element)

    new_sentence = ' '.join(reverse_array)
    return new_sentence

# Your task is to write a function that takes a string and return a new string with all vowels removed.

def disemvowel(string_):
    remove = ['a','e','i','o','u','A','E','I','O','U']
    for element in string_:
        if element in remove:
            string_ = string_.replace(element, "")
    return string_

# or

import re

def disemvowel(string_):
    return re.sub(r'[aeiouAEIOU]', '', string_)


# Implement the function unique_in_order which takes as argument a sequence 
# and returns a list of items without any elements with the same value next 
# to each other and preserving the original order of elements.

def unique_in_order(sequence):
    array = []
    for i in range(len(sequence)):
        prev = i - 1
        if i == 0 or sequence[i] != sequence[prev]:
            array.append(sequence[i])

    return array

# The Western Suburbs Croquet Club has two categories of membership, Senior and Open. 
# They would like your help with an application form that will tell prospective members which category they will be placed.

# To be a senior, a member must be at least 55 years old and have a handicap greater than 7. 
# In this croquet club, handicaps range from -2 to +26; the better the player the lower the handicap.

def open_or_senior(data):
    output = []
    for element in data:
        age = element[0]
        handicap = element[1]
        if age >= 55 and handicap > 7:
            output.append("Senior")
        else:
            output.append("Open")
    return output

# or

def openOrSenior(data):
  return ["Senior" if age >= 55 and handicap >= 8 else "Open" for (age, handicap) in data]


# Complete the function that accepts a string parameter, 
# and reverses each word in the string. All spaces in the string should be retained.
def reverse_words(text):
    array = text.split(' ')
    reverse_array = []
    for element in array:
        reverse = element[::-1]
        reverse_array.append(reverse)

    new_sentence = ' '.join(reverse_array)
    return new_sentence

# or

def reverse_words(str):
    return ' '.join(s[::-1] for s in str.split(' '))

# Complete the method/function so that it converts dash/underscore delimited words into camel casing. 
# The first word within the output should be capitalized only if the original word was capitalized 
# (known as Upper Camel Case, also often referred to as Pascal case). The next words should be always capitalized.
def to_camel_case(text):
    words = text.replace('-', '_').split('_')
    # Capitalize the first word if it was originally capitalized
    camel_case = words[0] + ''.join(word.capitalize() for word in words[1:])
    return camel_case

# Complete the solution so that the function will break up camel casing, using a space between words.
def solution(s):
    res = ""
    for i in range(len(s) - 1):
        res += s[i]
        if s[i + 1].isupper():
            res += " "

    res += s[-1]
    return res

File: codeWars.java
// convert an integer to binary and count how many 1's are in it's binary eqivalent
public class BitCounting {

	public static int countBits(int n){
		String binary = Integer.toBinaryString(n);
    int count = 0;
     
    for (int i = 0; i < binary.length(); i++) {
      if (binary.charAt(i) == '1') {
        count++;
      }
    }
    return count;
      
	}
	
}

// As the name may already reveal, it works basically like a Fibonacci, 
// but summing the last 3 (instead of 2) numbers of the sequence to generate the next.
//  And, worse part of it, regrettably I won't get to hear non-native Italian speakers trying to pronounce it :(
public class Xbonacci {

    public double[] tribonacci(double[] s, int n) {
        if (n == 0) {
            return new double[0];
        }


        double[] result = new double[n];
      
        for (int i = 0; i < 3 && i < n; i++) {
              result[i] = s[i];
          }
      
        for (int i = 3; i < n; i++) {
          result[i] = result[i - 1] + result[i - 2] + result[i - 3];
        }
      
      return result;

    }
  }

  // Complete the function that accepts a string parameter, 
  // and reverses each word in the string. All spaces in the string should be retained.

public class Kata
{
  public static String reverseWords(final String original)
  {
    String[] words = original.split(" ");
    //string builder is a mutable string of characters     
    StringBuilder reversedString = new StringBuilder();
    
    for (String word: words) {
      StringBuilder reversedWord = new StringBuilder(word).reverse();
      reversedString.append(reversedWord).append(" ");
    }
    
    return reversedString.toString().trim();
  }
}

// You probably know the "like" system from Facebook and other pages. People can "like" blog posts, pictures or other items. 

// We want to create the text that should be displayed next to such an item.

// Implement the function which takes an array containing the names of people that like an item. 

// It must return the display text as shown in the examples:

// []                                -->  "no one likes this"
// ["Peter"]                         -->  "Peter likes this"
// ["Jacob", "Alex"]                 -->  "Jacob and Alex like this"
// ["Max", "John", "Mark"]           -->  "Max, John and Mark like this"
// ["Alex", "Jacob", "Mark", "Max"]  -->  "Alex, Jacob and 2 others like this"
class Solution {
    public static String whoLikesIt(String... names) {
        int length = names.length;
        if (length == 0) {
            return "no one likes this";
        } else if (length == 1) {
            return names[0] + " likes this";
        } else if (length == 2) {
            return names[0] + " and " + names[1] + " like this";
        } else if (length == 3) {
            return names[0] + ", " + names[1] + " and " + names[2] + " like this";
        } else {
           return names[0] + ", " + names[1] + " and " + (length - 2) + " others like this";
        }
    }
}

// Complete the solution so that it splits the string into pairs of two characters. 
// If the string contains an odd number of characters then it should replace the missing second character of the final pair with an underscore ('_').
public class StringSplit {
  public static String[] solution(String s) {
      int n = s.length();
      int resultLength = (n % 2 == 0) ? n/2 : n/2 + 1;
      String[] result = new String[resultLength];
      for (int i = 1, j = 0; i <= n; i += 2, j++) {
        if(i == n) {
          result[j] = s.substring(i-1) + "_";
        } else {
            String str = s.substring(i-1, i+1);
            result[j] = str;
        }

      }
    System.out.print(result);
    return result;
  }
}
